builder/solver.go

```diff
@@ -178,3 +182,18 @@ func newInlineCacheOpt() client.CacheOptionsEntry {
 		Type: "inline",
 	}
 }
+
+func parseImageNameAndAttrs(s string) (string, map[string]string, error) {
+	entries := strings.Split(s, ",")
+	imageName := entries[0]
+	attrs := make(map[string]string)
+	var err error
+	for _, entry := range entries[1:] {
+		pair := strings.Split(strings.TrimSpace(entry), "=")
+		if len(pair) != 2 {
+			err = errors.New("Export attributes couldn't be parsed")
```

Comment:
we need to return here, otherwise we will keep executing below and will try to read from `pair[1]` which might not exist.
```suggestion
			return "", "", errors.Errorf("failed to parse export attribute: expected a key=value pair while parsing %q", entry)
```